authenticator/src/main/java/com/amplifyframework/ui/authenticator/AuthenticatorViewModel.kt

```diff
@@ -121,15 +123,20 @@ internal class AuthenticatorViewModel(
         this.configuration = configuration
 
         viewModelScope.launch {
-            val authConfig = authProvider.getConfiguration()
-
-            if (authConfig == null) {
-                handleGeneralFailure(MissingConfigurationException())
-                return@launch
+            when (val authConfigResult = authProvider.getConfiguration()) {
+                is AuthConfigurationResult.Invalid -> {
+                    handleGeneralFailure(
+                        InvalidConfigurationException(authConfigResult.message, authConfigResult.cause)
+                    )
+                    return@launch
+                }
+                AuthConfigurationResult.Missing -> {
```

Comment:
```suggestion
                is AuthConfigurationResult.Missing -> {
```
nit for matching syntax, I'll still approve

Reply:
`Missing` is an object so we don't use `is`.